spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/dynamodb/DynamoDbAutoConfiguration.java

```diff
@@ -101,31 +131,61 @@ static class StandardDynamoDbClient {
 
 		@ConditionalOnMissingBean
 		@Bean
-		public DynamoDbClient dynamoDbClient(AwsClientBuilderConfigurer awsClientBuilderConfigurer,
+		public DynamoDbClient standardDynamoDbClient(AwsClientBuilderConfigurer awsClientBuilderConfigurer,
 				ObjectProvider<AwsClientCustomizer<DynamoDbClientBuilder>> configurer, DynamoDbProperties properties) {
 			return awsClientBuilderConfigurer
 					.configure(DynamoDbClient.builder(), properties, configurer.getIfAvailable()).build();
 		}
+	}
+
+	@Conditional(MissingDaxUrlCondition.class)
+	@Configuration(proxyBeanMethods = false)
+	static class StandardDynamoDbAsyncClient {
+
+		@ConditionalOnMissingBean
+		@Bean
+		public DynamoDbAsyncClient standardDynamoDbAsyncClient(AwsClientBuilderConfigurer awsClientBuilderConfigurer,
+				ObjectProvider<AwsClientCustomizer<DynamoDbAsyncClientBuilder>> configurer,
+				DynamoDbProperties properties) {
+			return awsClientBuilderConfigurer
+					.configure(DynamoDbAsyncClient.builder(), properties, configurer.getIfAvailable()).build();
+		}
 
 	}
 
-	@ConditionalOnMissingBean
-	@Bean
-	public DynamoDbEnhancedClient dynamoDbEnhancedClient(DynamoDbClient dynamoDbClient) {
-		return DynamoDbEnhancedClient.builder().dynamoDbClient(dynamoDbClient).build();
+	@ConditionalOnClass(name = "software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClient")
+	@Configuration(proxyBeanMethods = false)
+	static class DynamoDbEnhancedClientConfiguration {
+		@ConditionalOnMissingBean
+		@ConditionalOnClass(DynamoDbEnhancedClient.class)
+		@Bean
+		public DynamoDbEnhancedClient dynamoDbEnhancedClient(DynamoDbClient dynamoDbClient) {
+			return DynamoDbEnhancedClient.builder().dynamoDbClient(dynamoDbClient).build();
+		}
+
+		@ConditionalOnMissingBean
+		@ConditionalOnClass(DynamoDbEnhancedAsyncClient.class)
+		@Bean
+		public DynamoDbEnhancedAsyncClient dynamoDbEnhancedAsyncClient(DynamoDbAsyncClient dynamoDbClient) {
+			return DynamoDbEnhancedAsyncClient.builder().dynamoDbClient(dynamoDbClient).build();
+		}
 	}
 
-	@ConditionalOnMissingBean(DynamoDbOperations.class)
-	@Bean
-	public DynamoDbTemplate dynamoDBTemplate(DynamoDbProperties properties,
-			DynamoDbEnhancedClient dynamoDbEnhancedClient, Optional<DynamoDbTableSchemaResolver> tableSchemaResolver,
-			Optional<DynamoDbTableNameResolver> tableNameResolver) {
-		DynamoDbTableSchemaResolver tableSchemaRes = tableSchemaResolver
+	@ConditionalOnClass(name = "io.awspring.cloud.dynamodb.DynamoDbOperations")
+	@Configuration(proxyBeanMethods = false)
+	static class DynamoDbTemplateConfiguration {
+		@ConditionalOnMissingBean(DynamoDbOperations.class)
+		@Bean
+		public DynamoDbTemplate dynamoDBTemplate(DynamoDbProperties properties,
```

Comment:
We should think about creating `DynamoDbAsyncTemplate`
This is just a note, not required for this PR.